src/NHibernate/Type/AbstractDateOnlyType.cs

```diff
@@ -0,0 +1,88 @@
+#if NET6_0_OR_GREATER
+using System;
+using System.Data.Common;
+using System.Globalization;
+using NHibernate.Engine;
+using NHibernate.SqlTypes;
+
+namespace NHibernate.Type
+{
+	/// <summary>
+	/// Base class for DateOnly types.
+	/// </summary>
+	[Serializable]
+	public abstract class AbstractDateOnlyType<TParameter> : PrimitiveType
+	{
+		protected AbstractDateOnlyType(SqlType sqlType) : base(sqlType)
+		{
+		}
+
+		public override string Name =>
+			GetType().Name.EndsWith("Type", StringComparison.Ordinal)
+				? GetType().Name[..^4]
+				: GetType().Name;
+
+		public override System.Type ReturnedClass => typeof(DateOnly);
+
+		public override System.Type PrimitiveClass => typeof(DateOnly);
+
+		public override object DefaultValue => DateOnly.MinValue;
+
+		/// <inheritdoc />
+		public override object Get(DbDataReader rs, int index, ISessionImplementor session)
+		{
+			try
+			{
+				return GetDateOnlyFromReader(rs, index, session);
+			}
+			catch (Exception ex) when (ex is not FormatException)
+			{
+				throw new FormatException(string.Format("Input string '{0}' was not in the correct format.", rs[index]), ex);
```

Comment:
What tells us the input is a string?

```suggestion
				throw new FormatException(string.Format("Input '{0}' was not convertible to DateOnly.", rs[index]), ex);
```